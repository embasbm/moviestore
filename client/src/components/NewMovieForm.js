import React, { Component } from 'react';
import { Panel, ControlLabel, FormGroup, FormControl, Button } from 'react-bootstrap';

class NewMovieForm extends Component {
  constructor(props) {
    super(props)
    this.titleInput = React.createRef();
    this.textInput = React.createRef();
    this.categoryId = React.createRef();
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit(e) {
    e.preventDefault()
    this.props.onNewMovie(this.titleInput.value, this.textInput.value, this.categoryId.value)
    this.titleInput.value = ''
    this.textInput.value = ''
    this.categoryId.value = ''
    this.titleInput.focus()
  }

  render() {
    return (
      <Panel bsStyle="primary">
        <Panel.Body>
          <form onSubmit={this.handleSubmit}>
            <FormGroup>
              <ControlLabel>Title</ControlLabel>
              <FormControl
                inputRef={ref => {this.titleInput = ref;}}
                id="movieTitle"
                type="text"
                label="Movie title"
                placeholder="Title..."
              />
              <ControlLabel>Text</ControlLabel>
              <FormControl
                inputRef={ref => { this.textInput = ref; }}
                id="movieText"
                type="text"
                label="Movie text"
                componentClass="textarea"
                placeholder="Text..."
              />
              <ControlLabel>Category</ControlLabel>
              <FormControl
                componentClass="select"
                placeholder="select"
                inputRef={ref => { this.categoryId = ref; }}
                >
                {this.props.categories.map(category => {
                  return (
                    <option key={category.id} value={category.id}>{category.name}</option>
                  )
                })}
              </FormControl>
            </FormGroup>
            <Button type="submit">Add Movie</Button>
          </form>
        </Panel.Body>
      </Panel>
    )
  }
}

export default NewMovieForm;
