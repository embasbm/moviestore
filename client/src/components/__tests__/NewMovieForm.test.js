import React from 'react';
import axios from "axios";
const ReactTestRenderer = require('react-test-renderer');
import NewMovieForm from '../NewMovieForm';

jest.mock('axios');

describe('NewMovieForm', () => {
  const resp = {
    data: []
  };

  const categories = [
    { id: 1, name: 'foo' }
  ]

  axios.get.mockImplementation(() => Promise.resolve(resp));

  it('Should compare NewMovieForm component with a snapshot', () => {
    const component = ReactTestRenderer.create(<NewMovieForm categories={categories} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  })
});
