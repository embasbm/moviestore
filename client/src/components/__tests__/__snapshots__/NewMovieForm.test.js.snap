// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMovieForm Should compare NewMovieForm component with a snapshot 1`] = `
<div
  className="panel panel-primary"
>
  <div
    className="panel-body"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="control-label"
          htmlFor={undefined}
        >
          Title
        </label>
        <input
          className="form-control"
          id="movieTitle"
          label="Movie title"
          placeholder="Title..."
          type="text"
        />
        <label
          className="control-label"
          htmlFor={undefined}
        >
          Text
        </label>
        <textarea
          className="form-control"
          id="movieText"
          label="Movie text"
          placeholder="Text..."
          type="text"
        />
      </div>
      <button
        className="btn btn-default"
        disabled={false}
        type="submit"
      >
        Add Movie
      </button>
    </form>
  </div>
</div>
`;
